presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -534,6 +560,15 @@ private int findLastSegmentStart(PagesHashStrategy pagesHashStrategy, Page page)
 
     private void closeAggregationBuilder()
     {
+        partialAggregationController.ifPresent(
+                controller -> controller.onFlush(
+                        inputBytesProcessed,
+                        inputRowsProcessed,
+                        aggregationBuilder instanceof SkipAggregationBuilder ? OptionalLong.empty() : OptionalLong.of(uniqueRowsProduced)));
```

Comment:
I guess since the aggregationBuilder is SkipAggregationBuilder only on request of the PartialAggregationController (per L601), can the state keeping of interpreting the uniqueRowsProduced as empty or valid-value be done by the PartialAggregationController ? ie, any way to simplify the caller logic by keeping some state in the callee ?